stdlib/public/core/StringLegacy.swift

```diff
@@ -33,14 +33,28 @@ extension String {
       self = count == 0 ? "" : repeatedValue
       return
     }
+    guard !repeatedValue.isEmpty else {
+      self = ""
+      return
+    }
 
-    // TODO(String performance): We can directly call appendInPlace
-    var result = String()
-    result.reserveCapacity(repeatedValue._guts.count &* count)
+    let repeatedValueGuts = repeatedValue._guts
+    let buffer: UnsafeMutableBufferPointer<UInt8> =
+      .allocate(capacity: repeatedValueGuts.count &* count)
+    defer { buffer.deallocate() }
+    var offset = 0
     for _ in 0..<count {
-      result += repeatedValue
+      let bufferRebased = UnsafeMutableBufferPointer(rebasing: buffer[offset...])
+      guard let copied = repeatedValueGuts.copyUTF8(into: bufferRebased) else {
+        fatalError("Buffer's capacity \(buffer.count)" +
```

Comment:
Unless I'm misunderstanding, this is not an error on the end user's part (the buffer capacity is not something under their control) and represents an internal logic error: probably an internal invariant failure.

Reply:
Fixed